src/store/modules/product/actions.ts

```diff
@@ -46,6 +46,39 @@ const actions: ActionTree<ProductState, RootState> = {
     // TODO Handle specific error
     return resp;
   },
+  async fetchProductDetails({ commit, state }, { productIds }) {
+    const cachedProductIds = Object.keys(state.cached);
+    const productIdFilter = productIds.reduce((filter: string, productId: any) => {
+      if(filter !== '') filter += ' OR '
```

Comment:
Move this condition in else part.
https://github.com/hotwax/bopis/blob/main/src/store/modules/product/actions.ts#L21

Reply:
No Sir, if we need to execute both lines we will put them in the same manner in which they are present, according to me. If you have any suggestions so please give me a hint.

Reply:
Sometimes, having this condition before will add an extra comma which will result in an error in searchProducts API, so we need to move this condition in the else part to only add a comma when we have a productId not present in cachedProductsIds